pkg/projectfile/projectfile.go

```diff
@@ -564,6 +564,39 @@ func (p *Project) LegacyCommitID() string {
 	return p.parsedURL.LegacyCommitID
 }
 
+func (p *Project) StripLegacyCommitID() error {
+	pf := NewProjectField()
+	if err := pf.LoadProject(p.Project); err != nil {
+		return errs.Wrap(err, "Could not load activestate.yaml")
+	}
+	pf.StripLegacyCommitID()
+	if err := pf.Save(p.path); err != nil {
+		return errs.Wrap(err, "Could not save activestate.yaml")
+	}
+
+	p.parsedURL.LegacyCommitID = ""
+	p.Project = pf.String()
+	return nil
+}
+
+// SetLegacyCommit sets the commit id within the current project file. This is done
+// in-place so that line order is preserved.
+// If headless is true, the project is defined by a commit-id only
```

Comment:
Nit: we don't have headless commits anymore, right? So this comment is superfluous.